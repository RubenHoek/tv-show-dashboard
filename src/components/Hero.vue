<script setup>
defineProps({
  titel: {
    type: String,
    required: true,
  },
  subtitle: {
    type: String,
    required: true,
  },
  danger: {
    type: Boolean,
    default: false,
  },
});
</script>

<template>
  <section class="hero is-small" :class="[danger ? 'is-danger' : 'is-primary']">
    <div class="hero-body">
      <p class="title">
        {{ titel }}
      </p>
      <p class="subtitle">
        {{ subtitle }}
      </p>
    </div>
  </section>
</template>
