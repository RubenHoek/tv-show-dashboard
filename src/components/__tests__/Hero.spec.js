import { it, describe, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Hero from "../Hero.vue";

import { createTestingPinia } from "@pinia/testing";

//Mock props
const titel = "Test title for Hero"
const subtitle = "Test subtitle for Hero"

describe.concurrent("Render the Hero", () => {
  it("renders the Hero", () => {
    const wrapper = mount(Hero, {
      props: { titel, subtitle },
      global: {
        plugins: [createTestingPinia({ createSpy: () => {} })],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
