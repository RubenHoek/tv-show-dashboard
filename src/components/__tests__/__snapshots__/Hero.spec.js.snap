// Vitest Snapshot v1

exports[`Render the Hero > renders the Hero 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <section
        class="hero is-small is-primary"
      >
        <div
          class="hero-body"
        >
          <p
            class="title"
          >
            Test title for Hero
          </p>
          <p
            class="subtitle"
          >
            Test subtitle for Hero
          </p>
        </div>
      </section>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {
          "$pinia": {
            "_a": [Circular],
            "_e": EffectScope {
              "active": true,
              "cleanups": [],
              "detached": true,
              "effects": [],
              "parent": undefined,
            },
            "_p": [
              [Function],
              [Function],
              [Function],
            ],
            "_s": Map {},
            "_testing": true,
            "app": [Circular],
            "install": [Function],
            "state": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {},
              "_value": {},
              "dep": undefined,
            },
            "use": [Function],
          },
        },
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {
        Symbol(pinia): {
          "_a": [Circular],
          "_e": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {},
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {},
            "_value": {},
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {
            "$pinia": {
              "_a": [Circular],
              "_e": EffectScope {
                "active": true,
                "cleanups": [],
                "detached": true,
                "effects": [],
                "parent": undefined,
              },
              "_p": [
                [Function],
                [Function],
                [Function],
              ],
              "_s": Map {},
              "_testing": true,
              "app": [Circular],
              "install": [Function],
              "state": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": undefined,
              },
              "use": [Function],
            },
          },
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {
          Symbol(pinia): {
            "_a": [Circular],
            "_e": EffectScope {
              "active": true,
              "cleanups": [],
              "detached": true,
              "effects": [],
              "parent": undefined,
            },
            "_p": [
              [Function],
              [Function],
              [Function],
            ],
            "_s": Map {},
            "_testing": true,
            "app": [Circular],
            "install": [Function],
            "state": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {},
              "_value": {},
              "dep": undefined,
            },
            "use": [Function],
          },
        },
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {
                "$pinia": {
                  "_a": [Circular],
                  "_e": EffectScope {
                    "active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [],
                    "parent": undefined,
                  },
                  "_p": [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "_s": Map {},
                  "_testing": true,
                  "app": [Circular],
                  "install": [Function],
                  "state": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "use": [Function],
                },
              },
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {
              Symbol(pinia): {
                "_a": [Circular],
                "_e": EffectScope {
                  "active": true,
                  "cleanups": [],
                  "detached": true,
                  "effects": [],
                  "parent": undefined,
                },
                "_p": [
                  [Function],
                  [Function],
                  [Function],
                ],
                "_s": Map {},
                "_testing": true,
                "app": [Circular],
                "install": [Function],
                "state": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "use": [Function],
              },
            },
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "danger": false,
            "subtitle": "Test subtitle for Hero",
            "titel": "Test title for Hero",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "danger": false,
            "subtitle": "Test subtitle for Hero",
            "titel": "Test title for Hero",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "danger": {
                "0": true,
                "1": true,
                "default": false,
                "type": [Function],
              },
              "subtitle": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "titel": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [
              "danger",
            ],
          ],
          "provides": {},
          "proxy": {
            "danger": false,
            "subtitle": "Test subtitle for Hero",
            "titel": "Test title for Hero",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Test title for Hero",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="title"
                    >
                      Test title for Hero
                    </p>,
                    "key": null,
                    "patchFlag": 1,
                    "props": {
                      "class": "title",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Test subtitle for Hero",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="subtitle"
                    >
                      Test subtitle for Hero
                    </p>,
                    "key": null,
                    "patchFlag": 1,
                    "props": {
                      "class": "subtitle",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="hero-body"
                >
                  <p
                    class="title"
                  >
                    Test title for Hero
                  </p>
                  <p
                    class="subtitle"
                  >
                    Test subtitle for Hero
                  </p>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "hero-body",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Test title for Hero",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="title"
                >
                  Test title for Hero
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "class": "title",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Test subtitle for Hero",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="subtitle"
                >
                  Test subtitle for Hero
                </p>,
                "key": null,
                "patchFlag": 1,
                "props": {
                  "class": "subtitle",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "dynamicProps": null,
            "el": <section
              class="hero is-small is-primary"
            >
              <div
                class="hero-body"
              >
                <p
                  class="title"
                >
                  Test title for Hero
                </p>
                <p
                  class="subtitle"
                >
                  Test subtitle for Hero
                </p>
              </div>
            </section>,
            "key": null,
            "patchFlag": 2,
            "props": {
              "class": "hero is-small is-primary",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "section",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Projecten/Mijn projecten/ABN Assignment/tv-show-dashboard/src/components/Hero.vue",
            "__name": "Hero",
            "components": {},
            "props": {
              "danger": {
                "0": true,
                "1": true,
                "default": false,
                "type": [Function],
              },
              "subtitle": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "titel": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <section
          class="hero is-small is-primary"
        >
          <div
            class="hero-body"
          >
            <p
              class="title"
            >
              Test title for Hero
            </p>
            <p
              class="subtitle"
            >
              Test subtitle for Hero
            </p>
          </div>
        </section>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
          "subtitle": "Test subtitle for Hero",
          "titel": "Test title for Hero",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Projecten/Mijn projecten/ABN Assignment/tv-show-dashboard/src/components/Hero.vue",
          "__name": "Hero",
          "components": {},
          "props": {
            "danger": {
              "0": true,
              "1": true,
              "default": false,
              "type": [Function],
            },
            "subtitle": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
            "titel": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {
              "$pinia": {
                "_a": [Circular],
                "_e": EffectScope {
                  "active": true,
                  "cleanups": [],
                  "detached": true,
                  "effects": [],
                  "parent": undefined,
                },
                "_p": [
                  [Function],
                  [Function],
                  [Function],
                ],
                "_s": Map {},
                "_testing": true,
                "app": [Circular],
                "install": [Function],
                "state": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "use": [Function],
              },
            },
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {
            Symbol(pinia): {
              "_a": [Circular],
              "_e": EffectScope {
                "active": true,
                "cleanups": [],
                "detached": true,
                "effects": [],
                "parent": undefined,
              },
              "_p": [
                [Function],
                [Function],
                [Function],
              ],
              "_s": Map {},
              "_testing": true,
              "app": [Circular],
              "install": [Function],
              "state": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": undefined,
              },
              "use": [Function],
            },
          },
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <section
          class="hero is-small is-primary"
        >
          <div
            class="hero-body"
          >
            <p
              class="title"
            >
              Test title for Hero
            </p>
            <p
              class="subtitle"
            >
              Test subtitle for Hero
            </p>
          </div>
        </section>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {
        "$pinia": {
          "_a": [Circular],
          "_e": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {},
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {},
            "_value": {},
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.41",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "danger": false,
    "subtitle": "Test subtitle for Hero",
    "titel": "Test title for Hero",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <section
    class="hero is-small is-primary"
  >
    <div
      class="hero-body"
    >
      <p
        class="title"
      >
        Test title for Hero
      </p>
      <p
        class="subtitle"
      >
        Test subtitle for Hero
      </p>
    </div>
  </section>,
}
`;
